task/mongo-optimization-task.js

```diff
@@ -229,26 +229,23 @@ async function task_3_1(db) {
         $lookup:
         {
             "from": "clientCriteria",
-            let: { v1: "$criteria_value" },
-            pipeline: [
+             let: {v1: "$criteria_value"},
+             pipeline: [
                 {
-                    $match: {
-                        "versions.initiativeId": ObjectId("58af4da0b310d92314627290"),
-                    }
-                },
+                    $match:{$and:[{"versions.initiativeId" : ObjectId("58af4da0b310d92314627290")},
+                    {$expr: {$eq: ["$value","$$v1" ]}}]
+                }
+               },
                 {
                     $project: {
                         "value": 1,
                         "label": 1,
                         "definition": 1,
                         "versions.definition": 1,
                     }
-                },
-                {
-                    $match: {
-                        $expr: { $eq: ["$value", "$$v1"] }
-                    }
-                },
+                },{
+                    $limit:1
```

Comment:
I added a limit and changed a index, but i don't know why it began to work more slowly. On my computer the test passed for 6840ms, in travis - 2552ms. I tried to add limit after first match, after second, at the end of pipeline - similar results.When i deleted limit, but left index, on my computer it was minimum 3050ms, in travis - 1777ms. When i brought back the old index, on my computerit was minimum 2999ms, in travis - 1711ms.
 Maybe am i doing it wrong?

Reply:
in travis result can be different depending on load of virtual machine. So +-1 sec is ok.
The best approach is execution plan investigating to understand if it work better for specific case.